### PR TITLE
Doxygen cleanup

### LINKED ISSUE #179: Use Doxygen effectively
**Abstract**

While Cantera has a long history of using Doxygen for docstrings, the generated website (and underlying organization) has not seen major updates for some time (little changed since Cantera 1.x). Based on a recent comment https://github.com/Cantera/enhancements/issues/178#issuecomment-1644858874 it is expected that Doxygen-generated web content will retain some role for the foreseeable future.

**Motivation**

There is a large amount of quality documentation contained in Doxygen docstrings. Unfortunately, inadquate organization, as well as the location of details within generated html files, makes this information hard to find. As a result, it is quite hard to get an overview of the organization of the source code, which creates a steep learning curve for anyone wanting to contribute new features implemented in C++.

Compared to the alternative (recreate documentation in Sphinx), updating Doxygen is comparably simple and should be considered the 'low-hanging-fruit' for improvements to Cantera documentation. 

**Description**

- [x] Update the main page ... Cantera/cantera#1516 / Cantera/cantera#1534
- [x] Add treeview and fix links ... Cantera/cantera#1535 / Cantera/cantera#1543
- [x] Switch Doxygen documentation theme / put details on top ... Cantera/cantera#1546
- [x] Add bibliography ... Cantera/cantera#1550
- [x] Add links to previous versions ... Cantera/cantera#1551
- [x] Standardize command style (`@` vs. `\` commands; most instances use the former, but there are some instances of the latter as well) ... Cantera/cantera#1557 and Cantera/cantera#1558; also Cantera/cantera#1565
- [x] Fix remaining warnings ... Cantera/cantera#1559
- [x] Ensure that autobrief works as intended ... Cantera/cantera#1560 / Cantera/cantera#1561
- [x] Update Doxygen style guide ... Cantera/cantera#1562
- [x] Sort BibTeX file alphabetically to reduce likelihood of merge conflicts ... Cantera/cantera#1567

Beyond, Doxygen documentation can be improved by increased use of `\defgroup` / `\addtogroup` / `\ingroup` as well as effective use of Doxygen markdown in docstrings and files (which replaces `*.dox`).

In addition, it is possible to move *advanced* content to Doxygen
- Consolidate YAML documentation ... Cantera/cantera#1548
- Move additional 'advanced' content from Sphinx as a long-term goal (?)

**Alternatives**

Use Sphinx + `breathe`/`exhale` or similar to render Doxygen XML content.

There is *no question* about Sphinx being the main tool of choice for the creation of user-facing web content. At the same time, it does not appear to be a good use of time to make C++ documentation accessible from Sphinx: Doxygen by itself has powerful - albeit currently underused - features, and Doxygen content can be easily cross-referenced by Sphinx using `sphinxcontrib-doxylink`. Maintaining a secondary 'Cantera Developer API' website - as has been done since at least Cantera 2.x - is a solution that requires minimal developer resources.

**References**

- #115
- #145
- #178

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

This PR collects final doxygen cleanup steps in the context of Cantera/enhancements#179

- Sort BibTeX file alphabetically to reduce future merge conflicts.
- Replace remaining single-letter Doxygen commands `\a`, `\b`, `\c`, `\e`, `\p` by `@b`, `@c` and `@e`
- Fix some LaTeX issues

**If applicable, fill in the issue number this pull request is fixing**

Closes Cantera/enhancements#179

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
@speth ... while there is clearly more potential for improvement, I am starting to see diminishing returns. Anything beyond can wait until after Cantera 3.0.

**Comment by ischoegl** [comment]
Rebased. Merge conflicts should (hopefully) wash out soon.

**Comment by ischoegl** [comment]
@speth ... resolved merge conflicts, so this is ready for a review.

File: include/cantera/thermo/IdealSolidSolnPhase.h [hunk 1]
```diff
@@ -89,7 +89,7 @@ class IdealSolidSolnPhase : public ThermoPhase
      * partial molar volume solution mixture with pure species phases which
      * exhibit zero volume expansivity:
      * @f[
-     * \hat s(T, P, X_k) = \sum_k X_k \hat s^0_k(T)  - \hat R \sum_k X_k log(X_k)
+     * \hat s(T, P, X_k) = \sum_k X_k \hat s^0_k(T)  - \hat R \sum_k X_k \log(X_k)
```
Comment (speth) [inline include/cantera/thermo/IdealSolidSolnPhase.h]: I think almost all of these instances of `\log` should actually be `\ln`. There are a few exceptions, though.

**Review by speth — CHANGES_REQUESTED** [review]
In the long run, I think I might prefer Markdown-style formatting over the Doxygen-specific `@e`, `@c` etc, but this is fine for now.

File: include/cantera/thermo/IdealSolidSolnPhase.h (hunk 1 above)
Comment (ischoegl) [reply include/cantera/thermo/IdealSolidSolnPhase.h]: Fair point. I believe all instances touched by this PR use `\ln`. As an aside, there are quite a few `\log` instances that are untouched by this PR, which I am leaving for Cantera/enhancements#6.

**Review by speth — APPROVED** [review]

**Pull request merged**